compiler/src/main/kotlin/dev/zacsweers/metro/compiler/fir/checkers/AggregationChecker.kt

```diff
@@ -65,7 +65,47 @@ internal object AggregationChecker : FirClassChecker(MppCheckerKind.Common) {
       if (classId in classIds.allContributesAnnotations) {
         val scope = annotation.resolvedScopeClassId() ?: continue
         val replaces = emptySet<ClassId>() // TODO implement
-
+        val checkIntoSet by lazy {
```

Comment:
nit: I have an `unsafeLazy` that acts like a simple memoize to skip the doubly-checked locking since this is never multithreaded